controllers/controller/workspacerouting/workspacerouting_controller.go

```diff
@@ -71,23 +73,31 @@ func (r *WorkspaceRoutingReconciler) Reconcile(req ctrl.Request) (ctrl.Result, e
 		return reconcile.Result{}, err
 	}
 
+	if instance.Annotations[config.WorkspaceExternalRoutingAnnotation] != "" {
+		return reconcile.Result{}, nil
+	}
+
 	// Check if the WorkspaceRouting instance is marked to be deleted, which is
 	// indicated by the deletion timestamp being set.
 	if instance.GetDeletionTimestamp() != nil {
 		reqLogger.Info("Finalizing WorkspaceRouting")
 		return reconcile.Result{}, r.finalize(instance)
 	}
 
-	if instance.Status.Phase == controllerv1alpha1.RoutingFailed {
-		return reconcile.Result{}, nil
-	}
-
 	solver, err := getSolverForRoutingClass(instance.Spec.RoutingClass)
 	if err != nil {
-		reqLogger.Error(err, "Could not get solver for routingClass")
```

Comment:
This is not correct, because there can be other error conditions than just an unknown solver.

Reply:
Good catch.

Reply:
That's what I get for throwing together a PR at the end of the day :smile: 

I'll fix it.